subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -113,6 +113,128 @@ In Gradle 8.0 and above, use the `AntlrSourceDirectorySet` source set extension
+This was an internal API, but may have been accessed by plugins.
+Supply a `FileResolver` to the overloaded version of this method instead.
+
+==== Groovydoc API Cleanup
+
+The deprecated `isIncludePrivate` property of the `Groovydoc` task type has been removed.
+Use the `access` property along with the `GroovydocAccess#PRIVATE` constant instead.
+
+==== JavaApplication API Cleanup
+
+The deprecated `mainClassName` property of the `JavaApplication` interface has been removed.
+Use the `mainClass` property instead.
+
+==== DefaultDomainObjectSet API Cleanup
+
+The deprecated `DefaultDomainObjectSet(Class)` constructor has been removed.
+This was an internal API, but may have been used by plugins.
+
+==== JacocoPluginExtension API Cleanup
+
+The deprecated `reportsDir` property of the `JacocoPluginExtension` has been removed.
+Use the `reportsDirectory` property instead.
+
+==== DependencyInsightReportTask API Cleanup
+
+The deprecated `legacyShowSinglePathToDependnecy` property of the `DependencyInsightReportTask` task type has been removed.
+Use the `showSinglePathToDependency` property instead.
+
+==== Report API Cleanup
+
+The deprecated `destination`, and `enabled` properties of the `Report` type have been removed.
+Use the `outputLocation` and `required` properties instead.
+
+==== JacocoMerge Task Removed
+
+The deprecated `JacocoMerge` task type has been removed.
+The same functionality is also available on the `JacocoReport` task.
+
+==== JavaExec API Cleanup
+
+The deprecated `main` property of the `JavaExec` task type has been removed.
+Use the `mainClass` property instead.
+
+==== AbstractExecTask API Cleanup
+
+The deprecated `execResult` getter property of the `AbstractExecTask` task type has been removed.
+Use the `executionResult` getter property instead.
+
+==== AbstractTestTask API Cleanup
+
+The deprecated `binResultsDir` property of the `AbstractTestTask` task type has been removed.
+Use the `binaryResultsDirectory` property instead.
+
+==== SourceDirectorySet API Cleanup
+
+The deprecated `outputDir` property of the `SourceDirectorySet` type has been removed.
+Use the `destinationDirectory` property instead.
+
+==== VersionCatalog API Cleanup
+
+The deprecated `findDependency(String)` method and `dependencyAliases` property of the `VersionCatalog` type have been removed.
+Use the `findLibrary(String)` method and `libraryAliases` property instead.
+
+The deprecated `alias(String)` method of the `VersionCatalogBuilder` type has been removed.
+Use the `library(String, String, String)` or `plugin(String, String)` methods instead.
+
+==== WorkerExecutor API Cleanup
+
+The deprecated `submit(Class, Action)` method of the `WorkerExecutor` interface has been removed.
+Instead, obtain a `WorkQueue` via the `noIsolation()`. `classLoaderIsolation()`, `processIsolation()`, methods and use the `submit(Class, Action)` method on the `WorkQueue` instead.
```

Comment:
```suggestion
Instead, obtain a `WorkQueue` via the `noIsolation()`, `classLoaderIsolation()`, and `processIsolation()`, methods and use the `submit(Class, Action)` method on the `WorkQueue` instead.
```

Reply:
Thanks for taking such a careful look at this.